typed-protocols/src/Network/TypedProtocol/Driver.hs

```diff
@@ -83,31 +85,37 @@ import Numeric.Natural (Natural)
 --
 runPeer
   :: forall ps (st :: ps) pr failure bytes m a .
-     (MonadThrow m, Exception failure)
-  => Codec ps failure m bytes
+     (MonadThrow m, Exception failure, Show (AnyMessage ps))
+  => Tracer m String
```

Comment:
Should use a different type IMO. A sum with each of the things that may be traced.

Reply:
What about `Tracer m Dynamic` ?

Reply:
Or `Tracer m Traceable`
where 
```
data Traceable = Str String
                          | Blk Block
                          | Msg ...
```
?

Reply:
https://github.com/input-output-hk/ouroboros-network/pull/298/commits/23b9bf52fefa676c5b1f44c1a6ec4fd69d617cea#diff-83b90bbd642add21a4351e069869662cR238